benchmarks/yamauchi_takei_2016_anelasticity/anelasticity_temperature.cc

```diff
@@ -0,0 +1,408 @@
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/global.h>
+#include "anelasticity_temperature.h"
+#include <aspect/material_model/interface.h>
+#include <aspect/initial_composition/interface.h>
+#include <aspect/adiabatic_conditions/interface.h>
+#include <aspect/simulator_access.h>
+#include <cmath>
+#include <algorithm>
+#include <functional>
+
+#include <boost/lexical_cast.hpp>
+#include <boost/math/tools/minima.hpp>
+
+namespace aspect
+{
+  namespace InitialTemperature
+  {
+    template <int dim>
+    AnelasticVs2T<dim>::AnelasticVs2T ()
+    {}
+
+    template <int dim>
+    void
+    AnelasticVs2T<dim>::initialize ()
+    {
+      Utilities::AsciiDataInitial<dim>::initialize(1);
+    }
+
+// set up Vs function that Brent minimization operates on
+    template <int dim>
+    double
+    AnelasticVs2T<dim>::
+    fVs (double x,
+         const double      depth,
+         const double      absolute_Vs,
+         const double      mu0,
+         const double      dmudT,
+         const double      dmudP,
+         const double      viscosity_prefactor,
+         const double      activation_energy,
+         const double      activation_volume,
+         const double      solidus_gradient,
+         const bool      use_original_model) const
+    {
+      return abs(yamauchi_takei_Vs(x,depth,mu0,dmudT,dmudP,viscosity_prefactor,activation_energy,activation_volume
+                                   ,solidus_gradient,use_original_model)-absolute_Vs);
+    }
+
+// set up volume change function that Brent minimization operates on
+    template <int dim>
+    double
+    AnelasticVs2T<dim>::
+    fdV (double x, const double bulk_modulus, const double bulk_modulus_pressure_derivative, const double pressure ) const
+    {
+      return abs((bulk_modulus*(3./2.)*(pow(x,7./3.)-pow(x,5./3.))*(1+(((3./4.)*
+                                                                        (bulk_modulus_pressure_derivative-4))*(pow(x,2./3.)-1))))-pressure);
+    }
+
+// set up initial temperature
+    template <int dim>
+    double
+    AnelasticVs2T<dim>::
+    initial_temperature (const Point<dim> &position) const
+    {
+      // initialize depth
+      double depth;
+      // if working in a box geometry, take y coordinate
+      if (this->get_geometry_model().has_curved_elements() == false)
+        {
+          depth = position(1);
+        }
+      // if working in a spherical geometry, take find depth from geometry model call
+      else
+        {
+          depth = this->get_geometry_model().depth(position);
+        }
+
+      // initialize temperature
+      double temperature;
```

Comment:
actually, you specifically *don't* initialize the variable here :-) You just declare it.